tools/accuracy_checker/accuracy_checker/evaluators/quantization_model_evaluator.py

```diff
@@ -166,19 +181,24 @@ def process_dataset(
         if check_progress:
             progress_reporter = ProgressReporter.provide('print', self.dataset.size)
 
-        for batch_id, (batch_annotation, batch_inputs, batch_identifiers) in enumerate(self.dataset):
+        for batch_id, (batch_input_ids, batch_annotation, batch_inputs, batch_identifiers) in enumerate(self.dataset):
             filled_inputs, batch_meta = self._get_batch_input(batch_inputs, batch_annotation)
-            batch_predictions = self.launcher.predict(filled_inputs, batch_meta, **kwargs)
+            batch_raw_predictions = self.launcher.predict(filled_inputs, batch_meta, **kwargs)
             if self.adapter:
                 self.adapter.output_blob = self.adapter.output_blob or self.launcher.output_blob
-                batch_predictions = self.adapter.process(batch_predictions, batch_identifiers, batch_meta)
+                batch_predictions = self.adapter.process(batch_raw_predictions, batch_identifiers, batch_meta)
+            else:
+                batch_predictions = batch_raw_predictions
 
             annotations, predictions = self.postprocessor.process_batch(batch_annotation, batch_predictions, batch_meta)
+            metrics_result = None
             if self.metric_executor:
-                self.metric_executor.update_metrics_on_batch(annotations, predictions)
+                metrics_result = self.metric_executor.update_metrics_on_batch(batch_input_ids, annotations, predictions)
+                if self.metric_executor.need_store_predictions:
+                    self._annotations.extend(annotations)
+                    self._predictions.extend(predictions)
 
-            self._annotations.extend(annotations)
-            self._predictions.extend(predictions)
+            output_callback(batch_raw_predictions, metrics_result=metrics_result)
```

Comment:
2 @eaidova:  sorry, the case `output_callback is None` is possible here? probably,  the check `if output_callback` is required?

Reply:
@LeonidBeynenson, thank you, fixed